lib/bolt/cli.rb

```diff
@@ -661,8 +661,9 @@ def show_targets
     end
 
     def list_groups
-      groups = inventory.group_names
-      outputter.print_groups(groups)
+      groups = inventory.group_names.sort
+      inventoryfile = config.inventoryfile || config.default_inventoryfile
```

Comment:
Hmm, it actually looks like this and the `target show` commands are missing `ENV[Bolt::Inventory::ENVIRONMENT_VAR]` as a possible source. I think the simplest way to reliably get the inventory source would be to have the inventory itself know where it comes from (since it can only have one source), so add a `source` instance var to `Bolt::Inventory::Inventory` then we can call `inventory.source` whenever we need this info.

Reply:
Opted to put this logic directly in the outputter. While storing this info on the inventory class does make sense, I don't think it makes things simpler here, since we need some way to tell if 1) the source is the environment variable or a file path and 2) check if the source exists and inventory was actually loaded from it and 3) if not, where we attempted to load it from. For now the simplest approach seems to be just checking if the env var is set in the outputter.

Reply:
I don't follow - the inventory object already does all of that checking for you, so you wouldn't have to do it here. If you set 'source' in `Bolt::Inventory.from_config`, pass that through to `create_version` (and include `source` from sources that call `create_version`). That seems reliable to me and wouldn't require any checking here.

Reply:
Maybe we could pass `nil` as the source to indicate inventory couldn't be found anywhere, though we'd also need a way to get the path to the default inventory file... so maybe we pass that along too when calling the outputter functions? :/ 

```
outputter.print_groups(
  groups,
  inventory.source,
  config.default_inventoryfile
)
```

Reply:
^ Went with the above approach.